### PR TITLE
Update Redis Client

### PR DESCRIPTION
**Pull request by Shaad7**
Signed-off-by: Shaad7 <abdullah.alshaad@appscode.com>

### CODE REVIEW

File: redis/kubedb_client_builder.go [hunk 1]
```diff
@@ -59,71 +57,116 @@ func (o *KubeDBClientBuilder) WithURL(url string) *KubeDBClientBuilder {
...
-	if err != nil {
-		return nil, fmt.Errorf("can't get the version from RedisVersion spec")
-	}
-
 	rdOpts := &rd.Options{
-		DialTimeout: 15 * time.Second,
-		IdleTimeout: 3 * time.Second,
-		PoolSize:    1,
-		Addr:        o.url,
+		DialTimeout:     15 * time.Second,
```
Comment (pkbhowmick) [inline redis/kubedb_client_builder.go:68]: Use constants

**Review by pkbhowmick — CHANGES_REQUESTED** [review]

File: redissentinel/kubedb_client_builder.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	if err != nil {
+		return "", err
+	}
+	clientPass := string(authSecret.Data[core.BasicAuthPasswordKey])
+	return clientPass, nil
+}
+
+func (o *KubeDBClientBuilder) getTLSConfig() (*tls.Config, error) {
+	var sec core.Secret
+	err := o.kc.Get(context.TODO(), client.ObjectKey{Namespace: o.db.Namespace, Name: o.db.CertificateName(api.RedisClientCert)}, &sec)
```
Comment (pkbhowmick) [inline redissentinel/kubedb_client_builder.go:117]: what about taking context in method param and then use in here?

**Review by pkbhowmick — APPROVED** [review]
LGTM!

**Review by heheh13 — APPROVED** [review]
LGTM!

**Pull request merged**